rclcpp/src/rclcpp/publisher.cpp

```diff
@@ -235,10 +234,16 @@ PublisherBase::operator==(const rmw_gid_t * gid) const
   return result;
 }
 
+rclcpp::mapped_ring_buffer::MappedRingBufferBase::SharedPtr
+PublisherBase::make_mapped_ring_buffer(size_t size) const
```

Comment:
If you move the `PublisherBase` into `publisher_base.hpp`, then please also make a `publisher_base.cpp` to match.